### PR TITLE
Removed `TextInputSpeech` memory leak

### LINKED ISSUE #2617: Remove Memory leak
**Expected behavior**
Shouldn't Leak

**Steps to reproduce the behavior:**
1. Install a brand new Custom app
2. Find Memory Leaks

**Environment**
- Version of Kiwix Android: 3.4
- Device: Real me
- OS version: 10
<details><summary>Logs</summary>

```
D/LeakCanary: ​
    ┬───
    │ GC Root: Global variable in native code
    │
    ├─ android.speech.tts.TextToSpeech$Connection$1 instance
    │    Leaking: UNKNOWN
    │    Retaining 1750918 bytes in 8000 objects
    │    Anonymous subclass of android.speech.tts.ITextToSpeechCallback$Stub
    │    ↓ TextToSpeech$Connection$1.this$1
    │                                ~~~~~~
    ├─ android.speech.tts.TextToSpeech$Connection instance
    │    Leaking: UNKNOWN
    │    Retaining 1750390 bytes in 7999 objects
    │    ↓ TextToSpeech$Connection.this$0
    │                              ~~~~~~
    ├─ android.speech.tts.TextToSpeech instance
    │    Leaking: UNKNOWN
    │    Retaining 1750365 bytes in 7998 objects
    │    mContext instance of org.kiwix.kiwixmobile.custom.CustomApp
    │    ↓ TextToSpeech.mInitListener
    │                   ~~~~~~~~~~~~~
    ├─ org.kiwix.kiwixmobile.core.main.KiwixTextToSpeech$tts$1 instance
    │    Leaking: UNKNOWN
    │    Retaining 1750026 bytes in 7985 objects
    │    Anonymous class implementing android.speech.tts.TextToSpeech$OnInitListener
    │    ↓ KiwixTextToSpeech$tts$1.this$0
    │                              ~~~~~~
    ├─ org.kiwix.kiwixmobile.core.main.KiwixTextToSpeech instance
    │    Leaking: UNKNOWN
    │    Retaining 1750010 bytes in 7984 objects
    │    context instance of org.kiwix.kiwixmobile.custom.main.CustomMainActivity with mDestroyed = true
    │    ↓ KiwixTextToSpeech.onAudioFocusChangeListener
    │                        ~~~~~~~~~~~~~~~~~~~~~~~~~~
    ├─ org.kiwix.kiwixmobile.core.main.-$$Lambda$CoreReaderFragment$dJ7yV3A9jJPxjfun6ekvjOK-1AA instance
    │    Leaking: UNKNOWN
    │    Retaining 12 bytes in 1 objects
    │    ↓ -$$Lambda$CoreReaderFragment$dJ7yV3A9jJPxjfun6ekvjOK-1AA.f$0
    │                                                               ~~~
    ╰→ org.kiwix.kiwixmobile.custom.main.CustomReaderFragment instance
    ​     Leaking: YES (ObjectWatcher was watching this because org.kiwix.kiwixmobile.custom.main.CustomReaderFragment
    ​     received Fragment#onDestroy() callback and Fragment#mFragmentManager is null)
    ​     Retaining 71806 bytes in 361 objects
    ​     key = ec514970-a68a-421d-a631-b68972454c50
    ​     watchDurationMillis = 6268
    ​     retainedDurationMillis = 1267
    
    METADATA
    
    Build.VERSION.SDK_INT: 29
    Build.MANUFACTURER: realme
    LeakCanary version: 2.5
    App process name: org.kiwix.kiwixcustomcustomexample
    Stats: LruCache[maxSize=3000,hits=5932,misses=81197,hitRate=6%]
    RandomAccess[bytes=4123041,reads=81197,travel=30590118522,range=21851238,size=25943630]
    Analysis duration: 5172 ms
I/BufferQueueConsumer: [org.kiwix.kiwixcustomcustomexample/leakcanary.internal.activity.LeakActivity#0](this:0x76f85f2800,id:10768,api:1,p:8514,c:783) acquireBuffer: slot 3 is dropped, handle=0x774bca4e80
```

### PR DESCRIPTION
**Pull request by gouri-panda**
Fixes #2617

### CODE REVIEW

**Comment by kelson42** [comment]
@gouri-panda This is a promising branch, do you plan to conplete it in a near future?

**Comment by gouri-panda** [comment]
Yes @kelson42 . I'll complete this soon.

**Comment by kelson42** [comment]
@gouri-panda Any news here? I would like to merge what as been fixed so far in 3.6.0 and close the PR.

**Comment by gouri-panda** [comment]
@kelson42 Sure :) sorry for this delay :(. I'll finish all my pending PRs.

**Comment by kelson42** [comment]
@gouri-panda This is the only memory leak in custom apps? Is that only in custom apps?

**Comment by gouri-panda** [comment]
@kelson42 I renamed the PR. It's not a good idea to fix in one PR. We'll find other leaks and fix it individually.

**Review by MohitMaliFtechiz — APPROVED** [review]

**Comment by kelson42** [comment]
@gouri-panda Are these memory leaks sepecific to custom apps?

**Comment by gouri-panda** [comment]
@kelson42 No, it's generic.

**Comment by kelson42** [comment]
Can you rephrase the ticket and PR then? This is misleading.

**Comment by gouri-panda** [comment]
>Can you rephrase the ticket and PR then? This is misleading.

The generic meaning it's happening in our test custom app, so we assume all the customs apps.

**Comment by kelson42** [comment]
No, I mean why the PR title is "Removed TextInputSpeech memory leaks from Custom app" when it should be "Removed TextInputSpeech memory leaks"?

**Review by MohitMaliFtechiz — APPROVED** [review]
approved
